### PR TITLE
Rel 5.7.17.v3

### PR DESCRIPTION
**Pull request by azhebel**

### CODE REVIEW

**Comment by azhebel** [comment]
Forgot to add issue number DOC-406: garbd is no longer included in flow control calculations

File: doc/source/manual/xtrabackup_sst.rst [hunk 1]
```diff
@@ -146,14 +163,27 @@ if :option:`encrypt` is set to ``1``.
...
+ [sst]
+ sockopt="retry=2,interval=3"
 
-.. note:: You can also enable SSL-based compression with :option:`sockopt`.
-   This can be used in place of the XtraBackup ``compress`` option.
+The previous example causes socat to try to connect three times
+(initial attempt and two retries with a 3-second interval between attempts).
+
+.. note:: Before |PXC| version 5.6.35,
+   the value must begin with a comma, for example::
```
Comment (kennt-percona) [inline doc/source/manual/xtrabackup_sst.rst]: Also versions of 5.7 before 5.7.17.v3 (or whatever we call this version)

File: doc/source/manual/xtrabackup_sst.rst [hunk 2]
```diff
@@ -218,8 +240,10 @@ in :file:`my.cnf` before you enable this option.
 
 .. option:: cpat
 
-Used to define the files that need to be deleted in the :term:`datadir`
-before running SST,
+    :Default: ``'.*\.pem$\|.*init\.ok$\|.*galera\.cache$\|.*sst_in_progress$\|.*\.sst$\|.*gvwstate\.dat$\|.*grastate\.dat$\|.*\.err$\|.*\.log$\|.*RPM_UPGRADE_MARKER$\|.*RPM_UPGRADE_HISTORY$'``
+
+Used to define the files
+that need to be deleted in the :term:`datadir` before running SST,
```
Comment (kennt-percona) [inline doc/source/manual/xtrabackup_sst.rst]: The description is incorrect.  This option is used to specify files that won't be deleted.  Everything else in the datadir will be deleted EXCEPT for these files.

File: doc/source/manual/xtrabackup_sst.rst [hunk 3]
```diff
@@ -301,6 +326,22 @@ To disable initial SST timeout, set ``sst-initial-timeout=0``.
    or there is a high network latency or network disturbance
    (which can cause donor selection to take longer than 100 seconds).
 
+.. option:: tmpdir
+
+   :Default: Empty
+   :Example: /path/to/tmp/dir
+
```
Comment (kennt-percona) [inline doc/source/manual/xtrabackup_sst.rst]: Do we want to say what version this was added?

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  instead of both at the same time.
+
+* PXC-692: Improved SST and IST log messages
+  for better readability and unification.
+
+* PXC-694: Excluded the ``garbd`` node from flow control calculations.
+
+* PXC-725: Added extra checks to verify that SSL files
+  (certificate, certificate authority, and key)
+  are compatible before openning connection.
```
Comment (kennt-percona) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst]: The correct spelling is "opening"

File: doc/source/wsrep-status-index.rst [hunk 5]
```diff
@@ -129,6 +129,12 @@ Number of ``FC_PAUSE`` events received since the last status query.
 
 Number of ``FC_PAUSE`` events sent since the last status query.
 
+.. variable:: wsrep_flow_control_status
```
Comment (kennt-percona) [inline doc/source/wsrep-status-index.rst:132]: Can we add the version that this was added?

File: doc/source/manual/xtrabackup_sst.rst [hunk 6]
```diff
@@ -318,13 +359,15 @@ introduced by ``wsrep_sst_xtrabackup-v2``
 * ``openbsd-netcat`` or ``socat`` for transfer.
   ``socat`` is a direct dependency of |PXC| and it is the default.
 
-* ``xbstream`` or ``tar`` for streaming. ``tar`` is default.
+* ``xbstream`` (default) or ``tar`` for streaming.
```
Comment (kennt-percona) [inline doc/source/manual/xtrabackup_sst.rst]: This should be:
xbstream or tar for streaming.   xbstream is the default.

File: doc/source/howtos/garbd_howto.rst [hunk 7]
```diff
@@ -5,18 +5,20 @@ Setting up Galera Arbitrator
...
-nodes).
-
-This document will show how to add Galera Arbitrator node to a already set up
-cluster.
+<http://galeracluster.com/documentation-webpages/arbitrator.html>`
+is a member of |Percona XtraDB Cluster| that is used for voting
+in case you have a small number of servers (usually two)
+and don't want to add any more resources.
+For example, arbitrator can be set up on existing application server
+or on load balancer.
```
Comment (kennt-percona) [inline doc/source/howtos/garbd_howto.rst]: This sentence is a little awkward:

For example, an arbitrator can be set up on an existing application server or on a load balancer.

**Comment by kennt-percona** [comment]
Also, noticed that "pxc_encrypt_cluster_traffic" is not included in the Configuration Reference in https://www.percona.com/doc/percona-xtradb-cluster/LATEST/configure.html

File: doc/source/howtos/garbd_howto.rst (hunk 7 above)
Comment (kbauskar) [reply doc/source/howtos/garbd_howto.rst]: Actually it is just to given an hint that garbd doesn't need a dedicated server. Should we revamp it to suggest that garbd doesn't need a dedicated server but can be installed on machine that has other application running. (Please ensure machine has good n/w connectivity).

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+* Galera Replication library 3.20
+
+* wsrep API version 29
+
+All Percona software is open-source and free.
+Details of this release can be found in the
+`5.7.17-29.20 milestone on Launchpad
+<https://launchpad.net/percona-xtradb-cluster/+milestone/5.7.17-29.20>`_.
+
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst:26]: Let's add a special section above

Important feature:
-----------------------

PXC has been tuned for performance and to scale well with increasing workload threads. There is huge uplift in performance ranging from 3x-10x.

(Revamp the wording as you see fit keeping the main essence)

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  in :ref:`pxc-strict-mode`.
+
+* :jirabug:`PXC-782`: Updated ``xtrabackup-v2`` script
+  to use the :option:`tmpdir` option
+  (if it is set under ``[sst]``, ``[xtrabackup]`` or ``[mysqld]``,
+  in that order).
+
+* :jirabug:`PXC-783`: Improved the wsrep stage framework.
+
+* :jirabug:`PXC-784`: Fixed the ``pc.recovery`` procedure to abort
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst:84]: This has external lp bug id. Please quote it.
Check if other JIRA issue too has external bug id. Please quote them too.

File: doc/source/verify.rst [hunk 10]
```diff
@@ -45,3 +45,17 @@ and adding some records to the table on the first node.
       +---------+-----------+
       1 row in set (0.00 sec)
 
+Next Steps
+==========
+
+* Consider installing ProxySQL_ on client nodes to improve performance
```
Comment (kbauskar) [inline doc/source/verify.rst]: "Consider installing ProxySQL on client nodes to improve performance ..."

Not sure who validated this claim.
Please double-check.

File: doc/source/conf.py [hunk 11]
```diff
@@ -54,7 +54,7 @@
 # The short X.Y version.
 version = '5.7'
 # The full version, including alpha/beta/rc tags.
-release = '5.7.17-27.20'
+release = '5.7.17-29.20'
```
Comment (kbauskar) [inline doc/source/conf.py:57]: Why no -3 here ?

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst [hunk 12]
```diff
@@ -0,0 +1,118 @@
+.. rn:: 5.7.17-29.20
+
+===================================
+Percona XtraDB Cluster 5.7.17-29.20
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst:4]: Do we need -3 ? is actually a RE tracking.

File: doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+* Added support for passing the XtraBackup buffer pool size
+  with the ``use-memory`` option under ``[xtrabackup]``
+  and the ``innodb_buffer_pool_size`` option under ``[mysqld]``
+  when the ``--use-memory`` option is not passed
+  with the ``inno-apply-opts`` option under ``[sst]``.
+
+* Added the :variable:`wsrep_flow_control_status` variable
+  to indicate if node is in flow control (paused).
+
+* Fixed gcache page cleanup not triggering
```
Comment (kbauskar) [inline doc/source/release-notes/Percona-XtraDB-Cluster-5.7.17-29.20.rst:66]: jira number of the fixes above went missing.

**Review by kbauskar — APPROVED** [review]
No major comment that needs re-review. Some minor comment. Please fix them and then good to go.

**Pull request merged**